frontends/web/src/utils/transport-common.ts

```diff
@@ -14,11 +14,13 @@
  * limitations under the License.
  */
 
+import { TPayload } from '../api/socket';
```

Comment:
something is wrong here - api types are leaking into the generic transport-common.ts. 

imo in general, the socket files and functions are not very spread out and hard to follow. 

would it make sense to first change every use of `apiWebsocket` to `event-legacy.ts:subscribe` or `event.ts:apiSubscribe`? then the types also don't have to be combined into socket.ts and all the switch cases are reduced/removed. wdyt?